### ISSUE TITLE
Can't add new platforms

### ISSUE THREAD

**Issue by gabrielclima** [tsuru/tsuru#1936]
Hi, guys! 

Im trying to add a platform to start my tests on Tsuru (im fixing [my pr](https://github.com/tsuru/tsuru/pull/1817)). 

when i run `./tsuru platform-add python` 
return 

```sh
panic: runtime error: index out of range

goroutine 38 [running]:
github.com/tsuru/tsuru-client/vendor/github.com/Nvveen/Gotty.readTermInfo(0xc4201ddc60, 0x1e, 0x0, 0x0, 0x0)
	/home/gabriel/go/src/github.com/tsuru/tsuru-client/vendor/github.com/Nvveen/Gotty/gotty.go:232 +0xaff
github.com/tsuru/tsuru-client/vendor/github.com/Nvveen/Gotty.OpenTermInfo(0xc42000c245, 0xe, 0xc42000c245, 0xe, 0xc42002ce50)
	/home/gabriel/go/src/github.com/tsuru/tsuru-client/vendor/github.com/Nvveen/Gotty/gotty.go:45 +0x267
github.com/tsuru/tsuru-client/vendor/github.com/docker/docker/pkg/jsonmessage.DisplayJSONMessagesStream(0x2d75900, 0xc42000e520, 0x2d71d00, 0xc4201ddba0, 0x1, 0x1, 0x0, 0x0, 0x0)
	/home/gabriel/go/src/github.com/tsuru/tsuru-client/vendor/github.com/docker/docker/pkg/jsonmessage/jsonmessage.go:245 +0x5a3
github.com/tsuru/tsuru-client/vendor/github.com/tsuru/tsuru/io.(*SimpleJsonMessageFormatter).Format.func1(0xc4201e8db0, 0x2d71d00, 0xc4201ddba0, 0x1, 0x1)
	/home/gabriel/go/src/github.com/tsuru/tsuru-client/vendor/github.com/tsuru/tsuru/io/stream_writer.go:161 +0x95
created by github.com/tsuru/tsuru-client/vendor/github.com/tsuru/tsuru/io.(*SimpleJsonMessageFormatter).Format
	/home/gabriel/go/src/github.com/tsuru/tsuru-client/vendor/github.com/tsuru/tsuru/io/stream_writer.go:162 +0x418
```

Tsuru API logs shows
```sh
2018/02/11 04:07:27 ERROR: failure running HTTP request POST /1.0/platforms (500): error in docker node "http://node1:2375": write tcp 172.42.0.8:8080->172.42.0.1:48004: write: broken pipe
```

I  already looked at `github.com/Nvveen/Gotty/gotty.go:232` but, i have no idea what this function makes

Im still using branch `master` at moment

Can anyone help me?

**Comment by gfleury** [comment]
@gabrielclima  Thats something anoying happening on tsuru-client

try, 
tsuru -v 2 platform-add python

This should work even with the error... I will take a look later.

**Comment by gabrielclima** [comment]
@gfleury im using [tsuru-client 1.4](https://github.com/tsuru/tsuru-client/releases/tag/1.5.0-rc1) now and works fine.

**Comment by ggarnier** [comment]
@gabrielclima I don't think we changed anything related to platforms between 1.5.0 rc1 and rc3. Which is your tsuru server version?

Do you get this error with other platforms?

**Comment by gabrielclima** [comment]
@ggarnier 1.5.0 rc3 has vendor upgrade, im looking for [github.com/Nvveen/Gotty/gotty.go](https://github.com/Nvveen/Gotty/blob/master/gotty.go), [github.com/tsuru/tsuru/io/stream_writer.go](https://github.com/tsuru/tsuru/blob/master/io/stream_writer.go) and [github.com/moby/moby/pkg/jsonmessage/jsonmessage.go](https://github.com/moby/moby/blob/master/pkg/jsonmessage/jsonmessage.go) history

> Do you get this error with other platforms?

yes, all others platforms, also using `tsuru platform-add your-platform-name --dockerfile https://raw.githubusercontent.com/tsuru/platforms/master/python/Dockerfile`

**Comment by ggarnier** [comment]
I can't reproduce this error. Can you give more information about your environment?

**Comment by gfleury** [comment]
I was facing it, I've updated tsuru-client and tsuru (api) to latest master branch, and now is working normal.

**Comment by pzduniak** [comment]
I assume that you guys already patched it, but I actually ran into it in my own project and the fix was quite simple - Docker uses a forked library using its original repo's name, per https://github.com/moby/moby/blob/master/vendor.conf#L148. If you correct your vendor tool to grab that fork instead, it all works fine.

THIS IS PLATFORM-RELATED! You won't reproduce this issue in all environments.

**Comment by arxdsilva** [comment]
Sorry guys, missclick on mobile

**Comment by gabrielclima** [comment]
> I was facing it, I've updated tsuru-client and tsuru (api) to latest master branch, and now is working normal.
Can you update both and try again @gabrielclima?

@gfleury i have already updated to `master` and was not solved

**Comment by gabrielclima** [comment]
> I assume that you guys already patched it, but I actually ran into it in my own project and the fix was quite simple - Docker uses a forked library using its original repo's name, per https://github.com/moby/moby/blob/master/vendor.conf#L148. If you correct your vendor tool to grab that fork instead, it all works fine.
>
> THIS IS PLATFORM-RELATED! You won't reproduce this issue in all environments.

i have updated the `gotty "github.com/Nvveen/Gotty"` import to `gotty "github.com/ijc/Gotty"` and it works fine now =)

this commit [github.com/ijc/Gotty/6018b68](https://github.com/ijc/Gotty/commit/6018b68f96b839edfbe3fb48668853f5dbad88a3) solve it

guys, i tryed to update the import and the vendor but the file [github.com/moby/moby/pkg/jsonmessage/jsonmessage.go](https://github.com/moby/moby/blob/master/pkg/jsonmessage/jsonmessage.go) is from docker repository and when i use `dep ensure`, dep reverts the changes to keep the original repository (`gotty "github.com/Nvveen/Gotty"`) because docker use this import with this name. Docker uses a vendor that has a trick to use an import with project name but from another fork repository. Dep has no this feature =| 

What we can make?

**Comment by gabrielclima** [comment]
> I can't reproduce this error. Can you give more information about your environment?

@ggarnier im using Ubuntu 17.10:
```
Distributor ID:	Ubuntu
Description:	Ubuntu 17.10
Release:	17.10
Codename:	artfu
```
and golang 1.8
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/gabriel/go"
GORACE=""
GOROOT="/usr/lib/go-1.8"
GOTOOLDIR="/usr/lib/go-1.8/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build233005855=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

**Comment by santak-rvbd** [comment]
@gabrielclima your answer https://github.com/tsuru/tsuru/issues/1936#issuecomment-370723173 solved the issue for me as well. I patched the file and did
```
$ cd ~/go/src/github.com/tsuru/tsuru-client
$ make build
$ cp ./bin/tsuru /usr/local/bin/tsuru
```
and things started working for me

**Comment by pzduniak** [comment]
https://golang.github.io/dep/docs/Gopkg.toml.html#source

Most likely should suffice in your case.

**Comment by ggarnier** [comment]
@gabrielclima fixed in tsuru-client 1.5.1. Thanks for reporting!
